File: src/Domain/Entities/Order.cs
using Domain.Common;

namespace Domain.Entities
{
	public record Order : BaseEntity
	{
		public int CustomerId { get; set; }
		public int ProductId { get; set; }
		public int Quantity { get; set; }

		public Customer Customer { get; set; }
		public Product Product { get; set; }
	}
}

File: src/Frontend/src/@types/order.ts
import { ICustomer } from "./customer";
import { IProduct } from "./product";

export interface IOrder {
  id: number;
  customerId: number;
  productId: number;
  quantity: number;

  customer?: ICustomer;
  product?: IProduct;
}

export type ICreateOrder = Pick<
  IOrder,
  "customerId" | "productId" | "quantity"
>;

export type IUpdateOrder = Pick<
  IOrder,
  "customerId" | "productId" | "quantity"
>;
